### PR TITLE
Add Env Variables for APM DC

### PR DESCRIPTION
**Pull request by qiyaod**
Add the following environment variables for kubernetes deployment.yaml for APM Node.js and Python data collector

          - name: APPLICATION_NAME
            value: "{{tag ' .Release.Name '}}"
          - name: IBM_APM_SERVER_URL
            valueFrom:
              secretKeyRef:
                name: apm-server-config
                key: ibm_apm_server_url
                optional: true
          - name: IBM_APM_KEYFILE
            valueFrom:
              secretKeyRef:
                name: apm-server-config
                key: ibm_apm_keyfile
                optional: true
          - name: IBM_APM_KEYFILE_PASSWORD
            valueFrom:
              secretKeyRef:
                name: apm-server-config
                key: ibm_apm_keyfile_password
                optional: true
          - name: IBM_APM_INGRESS_URL
            valueFrom:
              secretKeyRef:
                name: apm-server-config
                key: ibm_apm_ingress_url
                optional: true
          - name: IBM_APM_ACCESS_TOKEN
            valueFrom:
              secretKeyRef:
                name: apm-server-config
                key: ibm_apm_access_token
                optional: true

### CODE REVIEW

File: generators/kubernetes/templates/deployment.yaml [hunk 1]
```diff
@@ -40,4 +40,36 @@ spec:
           {{#each storageDeploys.env}}
           - name: {{@key}}
             value: {{{this}}}
-          {{/each}}
\ No newline at end of file
+          {{/each}}
+          - name: APPLICATION_NAME
+            value: "{{tag ' .Release.Name '}}"
```
Comment (Aerex) [inline generators/kubernetes/templates/deployment.yaml]: Please use `{{{ }}}` so that your `tag ' . Releas Name '` is properly renderede

Comment (katheris) [reply generators/kubernetes/templates/deployment.yaml]: Due to this commit that is no longer required: https://github.com/ibm-developer/generator-ibm-cloud-enablement/commit/29eb3f7438722fa40c223fb8852b3e36506b25b8

Comment (Aerex) [reply generators/kubernetes/templates/deployment.yaml]: @katheris right I forgot about that fix. Thanks for the heads up

Comment (katheris) [reply generators/kubernetes/templates/deployment.yaml]: We do need a test to verify that the `{{tag }}` handlebars section is being processed correctly during generation though

Comment (qiyaod) [reply generators/kubernetes/templates/deployment.yaml]: @Aerex @katheris thanks for help to review this PR.

The "{{tag ' .Release.Name '}}" was discussed and suggested by David Currie. 
 - name: APPLICATION_NAME
   value: "{{tag ' .Release.Name '}}"

And also this have been verified by the java. Here is the similar PR already committed long time ago for java:
https://github.com/ibm-developer/generator-ibm-cloud-enablement/pull/98

Comment (katheris) [reply generators/kubernetes/templates/deployment.yaml]: We need an actual test in the codebase to check the contents after generation, not just a manual verification.

Comment (Aerex) [reply generators/kubernetes/templates/deployment.yaml]: Also, rebase your commit afterwards

**Pull request merged**